### DISCUSSION TITLE
board: esp01_1m uart baud

### DISCUSSION THREAD

**Discussion by Enjoy-Combi** [echavet/MitsubishiCN105ESPHome#63] [Q&A] [answered]
Hi,

Thank you for your work. I currently have two HP working with esphome and esp01_1m board.
For that board it was needed to add baud_rate : 0 to the logger section. Still needed as it is an hardward limitation ?

**Comment by echavet** [comment]
Hello @Enjoy-Combi ,
I convert this to a discussion as it is not an issue...

**Reply by Enjoy-Combi** [reply]
Thank,

Tested and working good with baud_rate : 0 in logger.

**Comment by echavet** [comment, marked as answer]
Actually, the reason for setting baud_rate to 0 is that the ESP8266 has only one hardware UART, which is by default used for logs. This component 'needs' (though I've never tested it with a software UART) the hardware UART for reliability concerns. The idea is to free up the UART by disabling the logs, thereby allowing its use for communication between the component and the unit.

I would say yes, you still need to set the baud_rate to 0, as the documentation states:
`baud_rate (Optional, int): The baud rate for the serial UART port. Defaults to 115200. Set to 0 to disable logging via UART.`

However, if you want to preserve UART logging capabilities, it's possible on the ESP8266 by using UART1.

```yaml
logger:    
  level: DEBUG
  hardware_uart: UART1
```
This configuration means you don't have to set the baud_rate to 0, and you can still use the UART logging feature by connecting through a USB UART-TTL module to pin 2 (TX) of the ESP8266.

In summary, you must disable logging on UART0 because it's utilized for communication with the unit. However, you are free to test any combination you want, as long as you prevent UART logging from conflicting with the communication with the heat pump unit.

**Comment by skyw33** [comment]
@Enjoy-Combi can you provide your working yaml?  I'm also using esp01 and cannot get it to work.

**Comment by skyw33** [comment]
Nevermind, I had the tx and rx wires swapped between the cn105 and esp01.
